### ISSUE TITLE
1—recognising cats and dogs

### ISSUE THREAD

**Issue by datalass1** [datalass1/fastai#1]
Start of your fast.ai journey! In today’s lesson you’ll set up your deep learning server, and train your first image classification model (a convolutional neural network, or CNN), which will learn to distinguish dogs from cats nearly perfectly.

**Comment by datalass1** [comment]
Learn what is needed. 

the fastai library sits on top of pytorch written by facebook

**Comment by datalass1** [comment]
currently 26minutes in and stuck at AWS setup of jupyter notebook

**Comment by datalass1** [comment]
david perkins wrote Making Learning Whole, has much to say about this. The basic idea is to teach “the whole game”. 
e.g. take music. You don’t start out by studying the theory of harmonics and how strings vibrate and the circle of fifths. Instead, you would take someone to a concert or get them to start playing an instrument.

**Comment by datalass1** [comment]
Some of the take homes from lesson 1:
- top down approach to learning
- empirical rather than theoretical 
- Google Brain started 2012 with Geoffery Hinton and is used in almost every part of the business 

the more technical things:
- GPUs power based on GFLOP/sec and GPUs are about 10 times faster than CPUs. They are also cheaper!
- three things: infinitely flexible function, all purpose parameter, fast and scalable.
- 1. infinitely flexible function: 
neural network > consists of a number of linear layers interspersed with some non-linear layers > this is the universal approximation theorm > universal approximation theorm can solve any problem to arbitarily close accuracy as long as you add enough parameters. 
- 2. all purpose parameter (finding a way to fit the parameters):
gradient descent > for the different parameters, how good are they at solving my problem > following the surface of the los function to find the local minimum.
- 3. Fast and scalable:
there is increasing hidden layers > multiple linear scaling > DEEP LEARNING (a neural network with multiple hidden layers).
- The key piece of a convolutional neural network is convolution. A good example is http://setosa.io/ev/image-kernels
- most kernels in deep learning are 3x3
- really fantastic interactive book http://neuralnetworksanddeeplearning.com/chap4.html
- learning rate: taking SMALL steps for stochastic gradient descent 
- A great paper about what the convolutional layers are learning: Visualizing and Understanding Convolutional Networks, Matthew D. Zeiler, Rob Fergus: https://arxiv.org/pdf/1311.2901.pdf
- an epoch means going through the entire dataset of images and using each tiny we do a mini batch to improve the model a little bit using gradient descent. Printing of validation and loss at the end of each epoch. If you run too many epochs you could make the model worse by overfitting. 

Examples of deep learning:
- Google gmail auto reply generator. 
- Microsoft Skype realtime translator.
- Neural Doodle which will render a doodle as works by an artist.
- diagnosing lung cancer, if you can find lung nodules earlier from the CT scan there is 10 times higher probablity of survival. 

Finally:
- know how to use numpy and fastai libraries.
- shift tab is so useful for function documentation. 
- look at source code how you can do something using ??
